app/src/main/java/com/gh4a/activities/CommitActivity.java

```diff
@@ -225,7 +225,7 @@ private void loadCommit(boolean force) {
 
     private void loadComments(boolean force) {
         final RepositoryCommentService service =
-                ServiceFactory.get(RepositoryCommentService.class, force);
+                ServiceFactory.get(RepositoryCommentService.class, force, ApiHelpers.MAX_PAGE_SIZE);
```

Comment:
Is there any advantage of passing `MAX_PAGE_SIZE` here over using it as default value in the `get()` call not accepting a page size? Looking at the amount of calls you made use `MAX_PAGE_SIZE` it seems like it should be the new default.

Reply:
I know it's repetitive, but I'm very reluctant to make it the default for `ServiceFactory.get` because it would also affect all paged lists in the app, which are perfectly fine with GitHub's default of 30 - using `MAX_PAGE_SIZE` for those would only make the initial loading slower.
I agree that it should be the default for `ApiHelpers.PageIterator` methods though. I can try to move the Service creation inside those methods, so that they can set the `pageSize` on their own.

Reply:
That sounds probably a bit too complicating. I think a good middle ground could be renaming your new method to `ServiceFactory.getForPageIterator(Class<S> serviceClass, boolean bypassCache)` (or something among those lines) and inserting the page size there? That would make the intent clear and still centralize the number at one point. WDYT?

Reply:
I think it's fine. I've named the `ServiceFactory` method `getForFullPagedLists` because I didn't want to couple it too much with the concept of `PageIterator`, but let me know if you have other naming suggestions.